addons/account/views/account_account_views.xml

```diff
@@ -67,7 +67,7 @@
                                     <group>
                                         <field name="internal_type" invisible="1" readonly="1"/>
                                         <field name="internal_group" invisible="1" readonly="1"/>
-                                        <field name="currency_id" options="{'no_create': True}" groups="base.group_multi_currency"/>
+                                        <field name="currency_id" options="{'no_create': True}" groups="base.group_no_one"/>
```

Comment:
you probably needs both groups

Reply:
I tried, but then the field is visible for users with group_multi_currency even when not in debug mode